### PR TITLE
README.md: Add instructions for testing HAXM

### LINKED ISSUE #96: macOS Build Instructions need improvement
I encountered several issues when following the build instructions for macOS and loading my unsigned build.

## Issue 1: SDKs
The Xcode project file is not configured to use the SDK that I had--this is briefly mentioned but not explicitly stated. The version which are installed can be queried with `xcodebuild -showsdks`. For me, on macOS 10.12.6, 10.13 was installed.

## Issue 2: Issue with signed kernel extensions
Apparently macOS has a security feature which prevents installation of unsigned kernel extensions. This must be disabled when building for yourself. To do this, restart into recovery mode (hold `CMD + R` during reboot), then run terminal and execute `csrutil disable`.

## Issue 3: Permission issue with loading kernel extension
Apparently the kext will not be loaded without sufficient permissions, necessitating:
```bash
sudo chown -R root:wheel intelhaxm.kext
sudo chmod -R 700 intelhaxm.kext
```

## Issue 4: Symbol resolution failure during extension loading
I got an error message stating: `The following symbols are unresolved for this kext: ... ___memcpy_chk`
This can be resolved by adding a `-D_FORTIFY_SOURCE=0` macro definition to the project. [Source](https://github.com/mryraghi/tcpcrypt)

## Issue 5: Missing instructions for loading the kernel extension
The following command must be run:
```
sudo kextutil intelhaxm.kext
```

Then the following error messages may be displayed, despite the extension actually being loaded:

```
Diagnostics for intelhaxm.kext:
Code Signing Failure: not code signed
kext-dev-mode allowing invalid signature -67062 0xFFFFFFFFFFFEFA0A for kext "intelhaxm.kext"
kext signature failure override allowing invalid signature -67062 0xFFFFFFFFFFFEFA0A for kext "/path/to/haxm/darwin/hax_driver/com_intel_hax/build/Debug/intelhaxm.kext"
```

The extension can be unloaded with:

```
sudo kextunload intelhaxm.kext
```

Thanks!

### PR DESCRIPTION
**Pull request by raphaelning**
Make the document more useful by explaining how to test a driver
or kext built from source. The instructions for installing a
test-signed intelhaxm.sys will be added later after we make the
HaxmLoader tool available for download.

Fixes #96.

Signed-off-by: Yu Ning <yu.ning@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,13 +54,57 @@ Windows.
...
+   1. Reboot.
+1. Install the test certificate:
+   1. Copy `IntelHaxm.cer` from the build environment to the test environment
+(if the two are not the same). This file is generated alongside `IntelHaxm.sys`
+by the `Debug` build configuration.
+   1. In the test environment, open an *elevated* Command Prompt and run
+`certmgr.exe /add X:\path\to\IntelHaxm.cer /s /r localMachine root`
+
+### Loading and unloading the test driver
+TODO: HaxmLoader.exe usage
```
Comment (junxiaoc) [inline README.md]: I think we need to add how to load haxm driver here, or else they could build haxm driver and couldn't use it?

Comment (raphaelning) [reply README.md]: See the commit log. The plan is to release that tool first, so I can provide a download link here.

File: README.md [hunk 2]
```diff
@@ -96,7 +96,35 @@ by the `Debug` build configuration.
...
+stops and then deletes the service. 
+
+To load the test driver:
+1. Open an *elevated* Command Prompt.
+1. Make sure no other HAXM driver is loaded.
+   1. If `sc query intelhaxm` shows the `intelhaxm` service as `RUNNING`, you
+must stop it first: `sc stop intelhaxm`
+   1. Otherwise, unload the previously loaded test driver, if any:
+`HaxmLoader.exe -u`
+1. `HaxmLoader.exe -i X:\path\to\IntelHaxm.sys`
```
Comment (junxiaoc) [inline README.md]: Load test driver by "HaxmLoader.exe -i X:\path\to\IntelHaxm.sys"

File: README.md [hunk 3]
```diff
@@ -155,6 +163,19 @@ To unload the test kext:
 the signed one installed to `/Library/Extensions/`:
 `sudo kextload /Library/Extensions/intelhaxm.kext`
 
+### Viewing kext logs
+On macOS, HAXM debug output goes to the system log database, and can be
+retrieved at almost any time.
+
+* On OS X 10.10, HAXM log messages are written immediately to
+`/var/log/system.log`. They can be viewed conveniently using _Console.app_ or
+the `syslog` command.
```
Comment (junxiaoc) [inline README.md]: dmesg cmd could be used to dump mac kernel message, do we need to mention it?

Comment (raphaelning) [reply README.md]: Thanks! `dmesg` is pretty useful despite a few limitations.

**Pull request merged**